numpy/lib/npyio.py

```diff
@@ -1013,7 +1013,8 @@ def read_data(chunk_size):
                                  % line_num)
 
             # Convert each value according to its column and store
-            items = [conv(val) for (conv, val) in zip(converters, vals)]
+            items = ([*map(single_converter, vals)] if single_converter
+                     else [conv(val) for (conv, val) in zip(converters, vals)])
```

Comment:
Can you use an approach more like #19618 here? That way, the logic inside this helper becomes simpler, and as a bonus you only have to do the `if` statement once per file rather than once per line.

Reply:
done (there's no change in speed as the `if` check is probably negligible compared to the function call, but I agree it reads better).